client/src/pages/ProductDetails.tsx

```diff
@@ -1,5 +1,109 @@
-import { Link } from 'react-router-dom';
+import { toDollars } from '../library/to-dollars';
+import { useParams } from 'react-router-dom';
+import type { Product } from '../library/data';
+import { useEffect, useState, useContext } from 'react';
+import { closeContext } from '../components/NavDrawerCloseContext';
 
 export function ProductDetails() {
-  return <Link to={'/'}>wya?</Link>;
+  const { productId } = useParams();
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [product, setProduct] = useState<Product>();
+  const { closeNavDrawer } = useContext(closeContext);
+
+  useEffect(() => {
+    async function loadProduct(productId: number) {
+      try {
+        const response = await fetch(`/api/catalog/details/${productId}`);
+        if (!response.ok) throw new Error(`fetch Error ${response.status}`);
+        const result = await response.json();
+        setProduct(result);
+      } catch (error) {
+        setError(error);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    loadProduct(Number(productId));
+  }, [productId]);
+
+  if (isLoading) return <div>Loading...</div>;
+  if (error) {
+    return (
+      <div>
+        Error Loading catalog:{' '}
+        {error instanceof Error ? error.message : 'Unknown Error'}
+      </div>
+    );
+  }
+
+  if (!product) return null;
```

Comment:
Good!

Another way I do this, since if `isLoading` is false and `product` is undefined, then something very bad happened, is to use `if (error || !product)` up on line 31.